internal_workflow_testsuite.go

```diff
@@ -257,19 +258,34 @@ func (env *testWorkflowEnvironmentImpl) newTestWorkflowEnvironmentForChild(optio
 	return childEnv
 }
 
-func (env *testWorkflowEnvironmentImpl) setActivityTaskList(tasklist string, activityFns ...interface{}) {
+func (env *testWorkflowEnvironmentImpl) setActivityTaskList(
+	tasklist string,
+	ao RegisterActivityOptions,
+	activityFns ...interface{},
+) {
 	for _, activityFn := range activityFns {
-		env.testSuite.RegisterActivity(activityFn)
+		env.getHostEnv().RegisterActivityWithOptions(activityFn, ao)
```

Comment:
This was added before we were able to mock activity. With the ability to mock activity/workflow, we don't need that private host env any more. :)